### PR TITLE
Replace `removeChild` with `remove`

### PR DESCRIPTION
**Pull request by remcohaszing**
This replaces most uses of `parent.removeChild(child)` with `child.remove()`.

The two are almost equivalent. The only difference is that `parent.removeChild(child)` throws if the given node is not a child of the parent, whereas `child.remove()` never throws. There is no noticable performance difference. The only reason to use `removeChild` is to support Internet Explorer, but that’s no longer supported by Monaco editor.

I tested this in the Monaco playground, but not very thoroughly.

### CODE REVIEW

File: src/vs/editor/browser/config/charWidthReader.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ class DomCharWidthReader {
 		this._readFromDomElements();
 
 		// Remove the container from the DOM
-		targetWindow.document.body.removeChild(this._container!);
+		this._container!.remove();
```
Comment (lramos15) [inline src/vs/editor/browser/config/charWidthReader.ts:59]:
```suggestion
		this._container?.remove();
```

**Review by lramos15 — CHANGES_REQUESTED** [review]

File: src/vs/editor/browser/config/charWidthReader.ts (hunk 1 above)
Comment (remcohaszing) [reply src/vs/editor/browser/config/charWidthReader.ts:59]: I deliberately chose to keep the same functionality here, but I agree your suggestion is an improvement.

**Review by lramos15 — APPROVED** [review]
Thanks for the information and the contribution

File: src/vs/workbench/contrib/webview/browser/pre/index.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
 	<meta charset="UTF-8">
 
 	<meta http-equiv="Content-Security-Policy"
-		content="default-src 'none'; script-src 'sha256-bQPwjO6bLiyf6v9eDVtAI67LrfonA1w49aFkRXBy4/g=' 'self'; frame-src 'self'; style-src 'unsafe-inline';">
+		content="default-src 'none'; script-src 'sha256-noHVLQsurkONXmA3fcuAmcZ8UPYm/db88mhm9gAXcvk=' 'self'; frame-src 'self'; style-src 'unsafe-inline';">
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/webview/browser/pre/index.html:8]: @mjbvz is this change expected?

Comment (alexdima) [reply src/vs/workbench/contrib/webview/browser/pre/index.html:8]: Looks good, it matches what I computed for the <script> tag at https://emn178.github.io/online-tools/sha256.html 

image

**Review by alexdima — APPROVED** [review]

**Comment by bpasero** [comment]
@remcohaszing @lramos15 isnt this a bad change?

image

Did we review this change throughly?

**Comment by remcohaszing** [comment]
Yes, you just beat me to this comment. It’s a sloppy mistake. It happens twice, right before and after that `else` keyword.

File: src/vs/base/browser/ui/list/rowCache.ts [hunk 3]
```diff
@@ -13,14 +13,6 @@ export interface IRow {
 	templateData: any;
 }
```
Comment (bpasero) [inline src/vs/base/browser/ui/list/rowCache.ts:14]: @joaomoreno fyi

File: src/vs/workbench/contrib/splash/browser/partsSplash.ts [hunk 4]
```diff
@@ -110,8 +110,6 @@ export class PartsSplash {
 
 		// remove initial colors
 		const defaultStyles = mainWindow.document.head.getElementsByClassName('initialShellColors');
-		if (defaultStyles.length) {
-			mainWindow.document.head.removeChild(defaultStyles[0]);
-		}
+		defaultStyles?.[0].remove();
```
Comment (bpasero) [inline src/vs/workbench/contrib/splash/browser/partsSplash.ts:113]: @remcohaszing this is not a good change, what if the array is empty?

File: src/vs/base/browser/ui/selectBox/selectBoxCustom.ts [hunk 5]
```diff
@@ -520,12 +520,7 @@ export class SelectBoxList extends Disposable implements ISelectBoxDelegate, ILi
 		return {
 			dispose: () => {
 				// contextView will dispose itself if moving from one View to another
-				try {
-					container.removeChild(this.selectDropDownContainer); // remove to take out the CSS rules we add
-				}
-				catch (error) {
-					// Ignore, removed already by change of focus
-				}
+				this.selectDropDownContainer.remove(); // remove to take out the CSS rules we add
```
Comment (bpasero) [inline src/vs/base/browser/ui/selectBox/selectBoxCustom.ts:523]: Could this throw? why was it try/catch before?

File: src/vs/workbench/contrib/splash/browser/partsSplash.ts (hunk 4 above)
Comment (remcohaszing) [reply src/vs/workbench/contrib/splash/browser/partsSplash.ts:113]: This was indeed wrong. Fixed it in #214348.

File: src/vs/base/browser/ui/selectBox/selectBoxCustom.ts (hunk 5 above)
Comment (remcohaszing) [reply src/vs/base/browser/ui/selectBox/selectBoxCustom.ts:523]: No. `parent.removeChild(child)` throws if `child` is not a child of `parent`. `child.remove()` never throws. Aside from not needing a reference to the parent, that’s the only difference.

**Pull request merged**